### ISSUE TITLE
Support Java Lambda .zip deployment packages with dependencies

### ISSUE THREAD

**Issue by ianbrandt** [localstack/localstack#913]
In a `LocalstackDockerExtension` JUnit 5 test I'm able to upload a Java `RequestHandler<TestRequest, String>` Lambda in a [.zip deployment package](https://docs.aws.amazon.com/lambda/latest/dg/create-deployment-pkg-zip-java.html) to S3, and invoke it using the AWS Java SDK.

This works fine if `TestRequest` is a .class file directly in the zip (e.g. at `com/mypackage/TestRequest.class`).

If instead the `TestRequest` class file is in a .jar dependency in the lib folder (e.g. `lib/common.jar`), the invocation fails on a `NoClassDefFoundError` per the below.

Perhaps I've missed it, but I don't see anywhere [in the code](https://github.com/localstack/localstack/blob/master/localstack/services/awslambda/lambda_executors.py#L128) that adds `lib/` dependencies to the Java executor's classpath.

I can extract and publish a minimal test project that reproduces the issue if that would be of any help?

```
com.amazonaws.services.lambda.model.AWSLambdaException: Error executing Lambda function: Lambda process returned error status code: 1. Output:
Exception in thread "main" java.lang.NoClassDefFoundError: com/mypackage/TestRequest
	at com.mypackage.lamda.LocalStackHandler.handleRequest(LocalStackHandler.java:7)
	at cloud.localstack.LambdaExecutor.main(LambdaExecutor.java:100)
Caused by: java.lang.ClassNotFoundException: com.mypackage.TestRequest
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 2 more
 Traceback (most recent call last):
  File "/opt/code/localstack/localstack/services/awslambda/lambda_api.py", line 269, in run_lambda
    event, context=context, version=version, asynchronous=asynchronous)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 466, in execute
    process.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 462, in do_execute
    result = lambda_function(event, context)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_api.py", line 379, in execute
    event, context, handler=handler, main_file=main_file)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 487, in execute_java_lambda
    result, log_output = self.run_lambda_executor(cmd, asynchronous=asynchronous)
  File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 69, in run_lambda_executor
    (return_code, log_output))
Exception: Lambda process returned error status code: 1. Output:
Exception in thread "main" java.lang.NoClassDefFoundError: com/mypackage/TestRequest
	at com.mypackage.lamda.LocalStackHandler.handleRequest(LocalStackHandler.java:7)
	at cloud.localstack.LambdaExecutor.main(LambdaExecutor.java:100)
Caused by: java.lang.ClassNotFoundException: com.mypackage.TestRequest
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 2 more
 (Service: AWSLambda; Status Code: 500; Error Code: InternalFailure; Request ID: null)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1630)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1302)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1056)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:743)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:717)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:699)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:667)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:649)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:513)
	at com.amazonaws.services.lambda.AWSLambdaClient.doInvoke(AWSLambdaClient.java:2518)
	at com.amazonaws.services.lambda.AWSLambdaClient.invoke(AWSLambdaClient.java:2494)
	at com.amazonaws.services.lambda.AWSLambdaClient.executeInvoke(AWSLambdaClient.java:1423)
	at com.amazonaws.services.lambda.AWSLambdaClient.invoke(AWSLambdaClient.java:1398)
	at com.mypackage.lamda.LocalStackHandlerIT.testLocalStackHandler(LocalStackHandlerIT.java:80)
```

**Comment by ianbrandt** [comment]
I created a test case project that demonstrates the issue:

https://github.com/ianbrandt/localstack-test/

See in particular [`LocalStackLambdaZipDeploymentIT`](https://github.com/ianbrandt/localstack-test/blob/master/localstack-test-handler/src/test/java/com/ianbrandt/test/localstack/LocalStackLambdaZipDeploymentIT.java).

**Comment by whummer** [comment]
Thanks for reporting @ianbrandt . This should be fixed in #913 - please re-open this issue if the problem persists. Thanks
